src-plugins/manualRegistration/manualRegistration.cpp

```diff
@@ -196,49 +183,51 @@ template <typename PixelType, typename TransformType> int manualRegistrationPriv
         proc->output()->setData (result);
     }
 
-    return DTK_SUCCEED;
+    return medAbstractProcess::SUCCESS;
 }
 
 int manualRegistration::update(itkProcessRegistration::ImageType imgType)
 {
-    if(fixedImage().IsNull() || movingImages().isEmpty()
-            || movingImages()[0].IsNull())
-    {
-        displayMessageError("Either the fixed image or the moving image is empty");
-        return DTK_FAILURE;
-    }
-
-    if (imgType != itkProcessRegistration::FLOAT)
+    // Cast has been done in itkProcessRegistration
+    if (imgType == itkProcessRegistration::FLOAT)
     {
-        displayMessageError("Images type should be float");
-        return DTK_FAILURE;
+        return d->update<float>();
     }
 
-    return d->update<float>();
+    return medAbstractProcess::FAILURE;
 }
 
 itk::Transform<double,3,3>::Pointer manualRegistration::getTransform()
 {
-    return NULL;
+    itk::Transform<double, 3, 3>::Pointer outputTransform;
+    outputTransform = d->transform;
+    return outputTransform;
```

Comment:
These 3 lines can be simplified to ```return d->transform```.

Reply:
> /home/mathildemerle/Dev/medinria-superproject/medInria/src-plugins/manualRegistration/manualRegistration.cpp:205: erreur : could not convert '((manualRegistration*)this)->manualRegistration::d->manualRegistrationPrivate::transform' from 'itk::MatrixOffsetTransformBase<double, 3u>::Pointer {aka itk::SmartPointer<itk::MatrixOffsetTransformBase<double, 3u> >}' to 'itk::Transform<double, 3u, 3u>::Pointer {aka itk::SmartPointer<itk::Transform<double, 3u, 3u> >}'
>      return d->transform;
>                ^

That's why i put d->transform in `itk::Transform<double, 3, 3>::Pointer`

Reply:
You can use ```return d->transform.GetPointer()``` instead.